src/Uno.UWP/Storage/StorageFolder.cs

```diff
@@ -241,6 +240,52 @@ internal StorageFolder(string name, string path)
 				await TryInitializeStorage();
 
 				Directory.Delete(this.Path, true);
-			});
+            });
+
+		internal async Task<IReadOnlyList<IStorageItem>> GetItemsTask()
+		{
+			var items = new List<IStorageItem>();
+
+			foreach (var folder in Directory.GetDirectories(this.Path))
+			{
+				items.Add(await StorageFolder.GetFolderFromPathAsync(folder));
+			}
+
+			foreach (var folder in Directory.GetFiles(this.Path))
+			{
+				items.Add(await StorageFile.GetFileFromPathAsync(folder));
+			}
+
+			return items.AsReadOnly();
+		}
```

Comment:
The propagation of the `CancellationToken` is important here. Enumerating the directories is also important as if the cancellation comes in effect, it can stop without enumerating everything (even in the underlying implementation).

```suggestion
		internal async Task<IReadOnlyList<IStorageItem>> GetItemsTask(CancellationToken ct)
		{
			var items = new List<IStorageItem>();

			foreach (var folder in Directory.EnumerateDirectories(this.Path))
			{
				items.Add(await StorageFolder.GetFolderFromPathAsync(folder).AsTask(ct));
			}

			foreach (var folder in Directory.EnumerateFiles(this.Path))
			{
				items.Add(await StorageFile.GetFileFromPathAsync(folder).AsTask(ct));
			}

			return items.AsReadOnly();
		}
```

Reply:
done